tests/client/test_client.py

```diff
@@ -173,3 +174,10 @@ def test_format_uri_trailing_slash(self):
     def test_format_uri_all_the_slashes(self):
         uri = "/ewca/2022/123/"
         assert self.client._format_uri_for_marklogic(uri) == "/ewca/2022/123.xml"
+
+    def test_user_agent(self):
+        user_agent = self.client.session.prepare_request(
+            Request("GET", "http://example.invalid")
+        ).headers["user-agent"]
+        assert user_agent.startswith("ds-caselaw-marklogic-api-client/")
+        assert "." in user_agent  # i.e. looks like some sort of semantic version
```

Comment:
can we change it so it's just an assertion on a regex pattern?

I think:
`assert re.match(r'^ds-caselaw-marklogic-api-client\/\d+\.\d+\.\d+$', user_agent)`
Could be what we want?

Sticking
`^ds-caselaw-marklogic-api-client\/\d+\.\d+\.\d+$` into the pattern in https://regex101.com/
and testing for example `ds-caselaw-marklogic-api-client/2.3.1` 

works.

If we expect a v before the version number, then we can just add that in the pattern